smt.go

```diff
@@ -357,31 +360,68 @@ func (smt *SparseMerkleTree) sideNodesForRoot(path []byte, root []byte) ([][]byt
 
 		if bytes.Equal(nodeHash, smt.th.placeholder()) {
 			// If the node is a placeholder, we've reached the end.
-			return reverseSideNodes(sideNodes), nodeHash, nil, nil
+			if getSiblingData {
+				siblingData, err = smt.ms.Get(sideNodes[i])
```

Comment:
This is not tested.
It's also a bit confusing, because we use `sideNodes` with both `append` and `[i]`.

Reply:
Thanks to the refactoring in #29, this is now both simplified and tested.